lib/checkother.cpp

```diff
@@ -1990,6 +2019,17 @@ void CheckOther::duplicateExpressionError(const Token *tok1, const Token *tok2,
                 "determine if it is correct.", CWE398, false);
 }
 
+void CheckOther::duplicateAssignExpressionError(const Token *tok1, const Token *tok2)
+{
+    const std::list<const Token *> toks = make_container< std::list<const Token *> >() << tok2 << tok1;
+
+    reportError(toks, Severity::style, "duplicateAssignExpression", 
+        "Both variables '" + tok1->str() + "' and '" + tok2->str() + "' are assigned the same expression.\n"
+        "Finding two variables that are assigned the same expression is suspicious and might "
```

Comment:
I don't think that the verbose message is very clear. It should mention what variables are assigned.